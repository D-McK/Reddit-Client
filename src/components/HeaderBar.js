import React from "react";
import RedditLogo from "../image/Reddit_logo_new.png";

export const HeaderBar = () => {
  return (
    <div className="header-bar">
      <img src={RedditLogo} alt="reddit logo" />
    </div>
  );
};
