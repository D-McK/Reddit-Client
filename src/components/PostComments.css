.comment-chain{
    display: flex;
    justify-content: center;
    min-width: 100%;
    flex-wrap: wrap;
    text-align: left;
}

.comment{
    list-style: none;
    width: 90%;
    background-color: #e9e7e7bf;
    margin: 5px auto;
    border-radius: 5px;
    font-weight: bold;
    line-height: 200%;
}