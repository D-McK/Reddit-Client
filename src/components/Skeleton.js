import React from "react";
import ContentLoader from "react-content-loader";

export const Skeleton = () => {
  return (
    <ContentLoader>
      <rect x="0" y="0" width="750" height="150" />
      <rect x="0" y="150" width="700" height="120" />
      <rect x="0" y="300" width="700" height="120" />
    </ContentLoader>
  );
};
