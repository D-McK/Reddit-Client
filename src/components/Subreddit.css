.post {
    display: flex;
    flex-wrap: wrap;
    max-width: 50%;
    margin: 30px auto;
    background-color: rgb(255, 255, 255);
    box-shadow: 5px 5px 10px rgb(164, 164, 167);
    border-radius: 5px;
    align-items: center;
    transition: 0.2s;
}

.post:hover {
    box-shadow: 10px 10px 10px rgb(164, 164, 167);
    transition: 0.2s;
}

.post li {
    width: 98%;
    font-size: 25px;
    font-weight: bold;
    list-style: none;
    padding: 10px 10px;
    background-color: #e9e7e7bf;
}
.post img {
    display: block;
    padding: 20px 0;
    object-fit: cover;  
    border-radius: 10px;
    overflow: hidden;
    margin: 0 auto;
}
.under-info{
    display: flex;
    width: 100%;
    justify-content: space-around;
    padding: 10px 0px;
}
.under-info p{
    font-weight: bold;
    
}
.chat-icon {
    padding-right: 10px;
    position: relative; 
    top: 4px;
}

.comments, .subreddit-from{
    transition: 0.2s;
    padding: 5px
}
.comments:hover, .subreddit-from:hover{
    transition: 0.2s;
    background-color: #ff4300;
    color: white;
    cursor: pointer;
    border-radius: 5px;
}