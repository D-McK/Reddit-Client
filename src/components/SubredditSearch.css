.search-container{
    display: flex;  
    background-color: rgb(255, 255, 255);
    box-shadow: 5px 5px 10px rgb(164, 164, 167);
    border-radius: 5px;
    justify-content: center;
}

.search-bar{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    max-width: 400px;
}
.search-bar p{
    font-size: 20px;
    margin-bottom: 0;
    padding-bottom: 0;
}
.search-bar .search-box{
    width: 100%;
}
.search-bar .submit{
    margin: 10px;
    font-size: 20px;
}

.search-container img{
    width: 300px;
    height: 100px;
}
button{
    box-shadow: 5px 5px 10px rgb(164, 164, 167);
    padding: 0 30px;
    transition: 0.2s;
    border-radius: 3px;
}
button:hover{
    box-shadow: 10px 10px 10px rgb(164, 164, 167);
    transition: 0.2s;
    background-color: #ff4300;
    background-position: 0% 100%;
    color: white;
    cursor: pointer;
}