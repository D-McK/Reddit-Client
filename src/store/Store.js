import { combineReducers, configureStore } from "@reduxjs/toolkit";
import subredditReducer from "./subredditSlice";
import postReducer from "./postSlice";

const rootReducer = combineReducers({
  subreddit: subredditReducer,
  comments: postReducer,
});

export default configureStore({
  reducer: rootReducer,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};
