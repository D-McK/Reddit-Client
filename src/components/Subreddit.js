import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Skeleton } from "./Skeleton";
import {
  fetchSubreddit,
  selectPosts,
  selectStatus,
  selectSubreddit,
} from "../store/subredditSlice";
import {
  checkPostSelected,
  clearComments,
  postSelected,
  selectedPostCommentsOpened,
} from "../store/postSlice";

import { BsChatLeft } from "react-icons/bs";
import "./Subreddit.css";
import { PostComments } from "./PostComments";

export const Subreddit = () => {
  const dispatch = useDispatch();

  const selectedPosts = useSelector(selectPosts);
  const selectedSubreddit = useSelector(selectSubreddit);
  const loadingStatus = useSelector(selectStatus);
  const isCommentsOpen = useSelector(selectedPostCommentsOpened);

  const loadingArr = [];
  for (let i = 0; i < 4; i++) {
    loadingArr.push(
      <div className="post" key={i}>
        <Skeleton />
      </div>
    );
  }

  useEffect(() => {
    dispatch(fetchSubreddit(selectedSubreddit));
    dispatch(clearComments());
  }, [selectedSubreddit, dispatch]);

  if (loadingStatus === "failed") {
    return (
      <div>
        <h1>Sorry that subreddit doesn't exist</h1>
      </div>
    );
  }

  return (
    <div>
      {loadingStatus === "loading" || selectedPosts.length === 0
        ? loadingArr
        : selectedPosts.map((post, index) => {
            return (
              <div className="post" key={index}>
                <li key={index}>{post.data.title}</li>
                {post.data.url.includes("i.redd.it") ? (
                  <img src={post.data.url} alt={post.data.subreddit} />
                ) : null}
                <div className="under-info">
                  <p
                    className="comments"
                    onClick={() => {
                      dispatch(postSelected(post.data.permalink));
                      dispatch(checkPostSelected());
                    }}
                  >
                    <BsChatLeft className="chat-icon" />
                    {`${post.data.num_comments} Comments`}
                  </p>
                  <p className="subreddit-from">r/{post.data.subreddit}</p>
                </div>
                {isCommentsOpen ? <PostComments /> : null}
              </div>
            );
          })}
    </div>
  );
};
