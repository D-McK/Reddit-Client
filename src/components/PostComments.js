import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { fetchComments, selectComments, selectPost } from "../store/postSlice";

import { Skeleton } from "./Skeleton";

import "./PostComments.css";

export const PostComments = () => {
  const dispatch = useDispatch();
  const selectedPost = useSelector(selectPost);
  const commentsForPost = useSelector(selectComments);

  useEffect(() => {
    dispatch(fetchComments(selectedPost));
  }, [dispatch, selectedPost]);

  if (commentsForPost.length === 0) {
    return (
      <div className="loading">
        <Skeleton />
      </div>
    );
  }

  return (
    <div className="comment-chain">
      {commentsForPost
        .filter((comment) => comment.data.author !== "[deleted]")
        .map((comment, index) => {
          return (
            <div className="comment">
              <p className="author" key={"author: " + index}>
                {comment.data.author}
              </p>
              <li key={index} className="comment-contents">
                {comment.data.body}
              </li>
            </div>
          );
        })}
    </div>
  );
};
