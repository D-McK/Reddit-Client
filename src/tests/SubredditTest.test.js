import React from "react";
import { Subreddit } from "../components/Subreddit";
import { mockData } from "./mock/mockData";

import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { setupStore } from "../store/Store";
import { act } from "react-dom/test-utils";
import {
  changeSearch,
  changeSubreddit,
  loadedPosts,
} from "../store/subredditSlice";

const store = setupStore({
  subreddit: { posts: mockData, subreddit: "all", status: "succeeded" },
});

describe("<Subreddit/>", () => {
  test("renders front page on open", async () => {
    const { container } = await act(async () =>
      render(
        <Provider store={store}>
          <Subreddit />
        </Provider>
      )
    );
    console.log(store.getState());
    console.log(container);
    await expect(container).toMatchSnapshot();
  });
  test("should default render the 'all' page", () => {
    const subreddit = store.getState().subreddit;
    expect(subreddit.subreddit).toBe("all");
  });

  test("reducer action should change subreddit", () => {
    store.dispatch(changeSubreddit("testing"));
    const subreddit = store.getState().subreddit;
    expect(subreddit.subreddit).toBe("testing");
  });

  test("reducer action should add search value to store", () => {
    store.dispatch(changeSearch("testing search"));
    const subreddit = store.getState().subreddit;
    expect(subreddit.typedsubreddit).toBe("testing search");
  });

  test("reducer action should use typed value if value not given in params", () => {
    store.dispatch(changeSearch("typed"));
    store.dispatch(changeSubreddit());
    const subreddit = store.getState().subreddit;
    expect(subreddit.subreddit).toBe("typed");
  });
  test("reducer action should update posts stored in state", () => {
    store.dispatch(loadedPosts(mockData[0]));
    const subreddit = store.getState().subreddit;
    expect(subreddit.posts.at(-1)).toBe(mockData[0]);
  });
});
