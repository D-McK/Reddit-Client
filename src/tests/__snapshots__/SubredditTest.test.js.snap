// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subreddit/> renders front page on open 1`] = `
<div>
  <div>
    <h1>
      Sorry that subreddit doesn't exist
    </h1>
  </div>
</div>
`;
